src/Symfony/Component/PropertyAccess/Mapping/Factory/ClassResolverTrait.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\PropertyAccess\Mapping\Factory;
+
+use Symfony\Component\PropertyAccess\Exception\InvalidArgumentException;
+
+/**
+ * Resolves a class name.
+ *
+ * @internal
+ *
+ * @author Kévin Dunglas <dunglas@gmail.com>
+ */
+trait ClassResolverTrait
+{
+    /**
+     * Gets a class name for a given class or instance.
+     *
+     * @param mixed $value
+     *
+     * @return string
+     *
+     * @throws InvalidArgumentException If the class does not exists
+     */
+    private function getClass($value)
```

Comment:
This looks like a utility method to me. Should we really add this via a trait? I'd rather put this into a static method of a final class somewhere.